torch/_dynamo/side_effects.py

```diff
@@ -143,6 +151,12 @@ def check_allowed_side_effect(self, item):
             return True
         if self.should_allow_side_effects_under_checkpoint():
             return True
+        if self.is_tracing_generator() and not is_side_effect_safe(
```

Comment:
I'm confused, I thought we only want to ban mutations in generators during reconstruct? This seems like it's generally banning mutation

Reply:
`is_tracing_generator` should actually be `is_reconstructing_generator`. The name is a bit misleading, as it gives the impression this code is executed whether one is reconstructing a generator or not. But `is_tracing_generator` is only True when `.reconstruct` is called.